data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -701,13 +704,22 @@ private <K> Object readProperty(RS rs, MappingContext<K> ctx, RuntimePersistentP
         }
         DataType dataType = prop.getDataType();
         Object result;
+        AttributeConverter<?, ?> converterForType = prop.getConverter(dialect);
```

Comment:
I think here the attribute converter can be of some type that can receive `ResultReader`, so in this case we can pass it and get the value